doc/client_setup.md

```diff
@@ -0,0 +1,199 @@
+## Setting up a new C2 client
+
+This document describes the details of setting up a new C2 client.
+
+### Overview
+
+A *client* represents a group of Users within the C2 system that share
+a data model. Existing examples include **GSA18F** and **NCR**.
+
+Within a *client* users may be assigned different roles. See the [Authorization](roles.md)
+document for details on roles.
+
+A *client* namespace is assigned to each User via a `client_slug` model value. The
+convention is that the `client_slug` is a downcased string matching the client model
+namespace. For example `ncr` for **NCR**.
+
+A User without an assigned `client_slug` may not create, edit, view or search Proposals,
+since the User will have no authorization context. The User may, however, login to the system.
+
+### Models
+
+The first thing to identify when adding a new client is the data schema for the client-specific
+extension to the Proposal. This extension is called *client_data*.
+
+Here is an example db migration for a new client called **Foo** with a client_data schema
+called **SpendingRequest**:
+
+```ruby
+class AddFooSpendingRequests < ActiveRecord::Migration
+  def change
+    create_table :foo_spending_requests do |t|
+      t.decimal  :amount
+      t.string   :project_title
+      t.datetime :created_at
+      t.datetime :updated_at
```

Comment:
super small nit but these two columns can be and are usually represented via `t.timestamps null: false`